tsdb/store.go

```diff
@@ -579,52 +554,103 @@ func (s *Store) shardsSlice() []*Shard {
 	return a
 }
 
-// DatabaseIndex returns the index for a database by its name.
-func (s *Store) DatabaseIndex(name string) *DatabaseIndex {
-	s.mu.RLock()
-	defer s.mu.RUnlock()
-	return s.databaseIndexes[name]
-}
-
-// Databases returns all the databases in the indexes
+// Databases returns the names of all databases managed by the store.
 func (s *Store) Databases() []string {
 	s.mu.RLock()
 	defer s.mu.RUnlock()
-	databases := make([]string, 0, len(s.databaseIndexes))
-	for db := range s.databaseIndexes {
-		databases = append(databases, db)
+
+	databases := make([]string, 0, len(s.databases))
+	for k, _ := range s.databases {
+		databases = append(databases, k)
 	}
 	return databases
 }
 
-// Measurement returns a measurement by name from the given database.
-func (s *Store) Measurement(database, name string) *Measurement {
+// DiskSize returns the size of all the shard files in bytes.
+// This size does not include the WAL size.
+func (s *Store) DiskSize() (int64, error) {
+	var size int64
+
 	s.mu.RLock()
-	db := s.databaseIndexes[database]
+	allShards := s.filterShards(nil)
 	s.mu.RUnlock()
-	if db == nil {
-		return nil
+
+	for _, sh := range allShards {
+		sz, err := sh.DiskSize()
+		if err != nil {
+			return 0, err
+		}
+		size += sz
 	}
-	return db.Measurement(name)
+	return size, nil
 }
 
-// DiskSize returns the size of all the shard files in bytes.  This size does not include the WAL size.
-func (s *Store) DiskSize() (int64, error) {
+func (s *Store) cardinalityEstimate(dbName string, getSketches func(*Shard) (estimator.Sketch, estimator.Sketch, error)) (int64, error) {
```

Comment:
Can you rename this to `estimateCardinality()`? It seems like it should be a verb.